src/main/java/edu/harvard/iq/dataverse/authorization/providers/oauth2/impl/MicrosoftOAuth2AP.java

```diff
@@ -0,0 +1,60 @@
+package edu.harvard.iq.dataverse.authorization.providers.oauth2.impl;
+
+import com.github.scribejava.apis.MicrosoftAzureActiveDirectory20Api;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.AbstractOAuth2AuthenticationProvider;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.logging.Logger;
+import java.io.StringReader;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+import edu.harvard.iq.dataverse.authorization.AuthenticatedUserDisplayInfo;
+
+/**
+ *
+ * @author
+ */
+public class MicrosoftOAuth2AP extends AbstractOAuth2AuthenticationProvider{
+
+    private static final Logger logger = Logger.getLogger(MicrosoftOAuth2AP.class.getCanonicalName());
+
+    public MicrosoftOAuth2AP(String aClientId, String aClientSecret){
+        this.id = "microsoft";
+        this.title = "Microsoft";
+        this.clientId = aClientId;
+        this.clientSecret = aClientSecret;
+        this.scope = Arrays.asList("openid User.Read");
```

Comment:
After reading some docs for Microsoft Graph, I don't think this scope setting makes sense.

It should be sufficient to use `User.Read`. We don't use OIDC here at all and that `openid` scope does not include any attributes - it will only allow the use to login.

See docs:
- https://docs.microsoft.com/en-us/graph/permissions-reference#user-permissions
- https://docs.microsoft.com/en-us/graph/permissions-reference#openid-permissions

Reply:
Yes is sufficient to use User.Read, We delete openid option.